minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java

```diff
@@ -16,16 +16,30 @@
 
 package com.google.cloud.tools.minikube;
 
+import com.google.cloud.tools.minikube.util.CommandExecutorFactory;
+import com.google.cloud.tools.minikube.util.MinikubeDockerEnvParser;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
 import org.gradle.api.Project;
 import org.gradle.api.provider.PropertyState;
 
 /** Minikube configuration extension. */
 public class MinikubeExtension {
   private final PropertyState<String> minikube;
 
-  public MinikubeExtension(Project project) {
+  private final CommandExecutorFactory commandExecutorFactory;
+
+  public MinikubeExtension(Project project, CommandExecutorFactory commandExecutorFactory) {
     minikube = project.property(String.class);
     setMinikube("minikube");
+
+    this.commandExecutorFactory = commandExecutorFactory;
+  }
+
+  public CommandExecutorFactory getCommandExecutorFactory() {
```

Comment:
Do we need this getter?

Reply:
Nope, I took it out. This was leftover from another attempt.